rule/rule.go

```diff
@@ -257,10 +257,10 @@ func scanExprs(defName string, stmt []bzl.Expr) (rules []*Rule, loads []*Load, f
 // If there is at least one matching file, a path will be returned by joining
 // dir and the first matching name. If there are no matching files, the
 // empty string is returned.
-func MatchBuildFileName(dir string, names []string, files []os.FileInfo) string {
+func MatchBuildFileName(dir string, names []string, ents []fs.DirEntry) string {
```

Comment:
Unfortunately this is a breaking change to public API. Can we add a new function and mark this one as deprecated? Or do you see a different way to prevent this while still reaping the performance benefits?

Reply:
oh good point, yeah let me add a new version instead